homeassistant/components/ista_ecotrend/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The ista Ecotrend integration."""
+
+from __future__ import annotations
+
+import logging
+
+from pyecotrend_ista.exception_classes import (
+    InternalServerError,
+    LoginError,
+    ServerError,
+)
+from pyecotrend_ista.pyecotrend_ista import PyEcotrendIsta
+from requests.exceptions import RequestException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import IstaCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+type IstaConfigEntry = ConfigEntry[IstaCoordinator]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: IstaConfigEntry) -> bool:
+    """Set up ista Ecotrend from a config entry."""
+
+    try:
+        ista = PyEcotrendIsta(
+            entry.data[CONF_EMAIL],
+            entry.data[CONF_PASSWORD],
+            _LOGGER,
+        )
```

Comment:
```suggestion
   ista = PyEcotrendIsta(
            entry.data[CONF_EMAIL],
            entry.data[CONF_PASSWORD],
            _LOGGER,
        )
    try:
```
only have code in `try` that can fail